pkg/kubelet/kubelet_pods.go

```diff
@@ -111,6 +112,30 @@ func (kl *Kubelet) makeDevices(pod *v1.Pod, container *v1.Container) ([]kubecont
 	return devices, nil
 }
 
+// makeBlockVolumes maps the raw block devices specified in the path of the container
+// Experimental
+func (kl *Kubelet) makeBlockVolumes(pod *v1.Pod, container *v1.Container, podVolumes kubecontainer.VolumeMap) ([]kubecontainer.DeviceInfo, error) {
+	var devices []kubecontainer.DeviceInfo
+	for _, device := range container.VolumeDevices {
+		vol, ok := podVolumes[device.Name]
+		if !ok || vol.BlockVolumeMapper == nil {
+			glog.Warningf("Block volume cannot be satisfied for container %q, because the volume is missing or the volume mapper is nil: %q", container.Name, device)
+			continue
```

Comment:
I'm a bit surprised we do this for filesystem volumes too.  Wouldn't it be a problem if we started a container with not all the volumes mounted, and the application wrote to the path expecting the data to be on a persisted volume?  I guess we don't create a path for it, so the application would get a PathNotFound error first.  But it's possible the user could mount the volume at an existing path too, in which case this would be an issue.

Reply:
I may be missing something?
But since this method is iterating VolumeDevices, there isn't a case that the pod volume is filesystem volume.

Reply:
Opened #53421 to handle the fs case